src/shapes/text.class.js

```diff
@@ -204,6 +204,22 @@
      */
     path:               null,
 
+    /**
+     * Offset amount for text path starting position
+     * Only used when text has a path
+     * @type Number
+     * @default
+     */
+    startOffset:               0,
+
+    /**
+     * Which side of the path the text should be drawn on.
+     * Only used when text has a path
+     * @type String
+     * @default
+     */
+    side:               "left",
```

Comment:
left/right can be up/down either. But why we do want this now?
textAlign and startoffset are big enough for a single PR.

Reply:
Side left/right is the name of the svg equivalent property so I figured it'd be good to be consistent. 
https://developer.mozilla.org/en-US/docs/Web/SVG/Attribute/side

The side logic is tied pretty closely to rtl support so I can open a separate pr for those if you think this too much for one pr.

Reply:
what happen if we just let the current logic run over a path during rtl mode?
is it completely garbage?

Reply:
ok, I see the problem. I was misunderstanding how right-to-left text works. I'll make some changes.

Reply:
My suggestion is to make textAlign on the path work. Cover it with visual tests as we can,
We can then looks like how it behaves with RTL text, and then if really needs some extra tweaks we can do that.